caput/profile.py

```diff
@@ -163,3 +168,156 @@ def __exit__(self, *args, **kwargs):
+        Returns
+        -------
+        cpu_times : `collections.namedtuple`
+            Same as `psutil.cpu_times`. Process CPU times since `start` was called in seconds.
+        cpu_percent :  float
+            Process CPU utilization since `start` was called as percentage. Can be >100 if multiple threads run on
+            different cores. See `PSUtil.cpu_count` for available cores.
+        disk_io : `collections.namedtuple`
+            Same as `psutil.io_counters` (on Linux) or `psutil.disk_io_counters` (on MacOS).
+            Difference since `start` was called.
+        memory : str
+            Difference of memory in use by this process since `start` was called. If negative,
+            less memory is in use now.
+        Raises
+        ------
+        RuntimeError
+            If stop was called before start.
+        """
+        stop_time = time.time()
+
+        # Get all stats at the same time
+        with self.oneshot():
+            cpu_times = self.cpu_times()
+            cpu_percent = self.cpu_percent()
+            memory = self.memory_full_info().uss
+            if psutil.MACOS:
+                disk_io = psutil.disk_io_counters()
+            else:
+                disk_io = self.io_counters()
+
+        if name not in self._start_cpu_times:
+            raise RuntimeError(
+                f"PSUtilProfiler.stop was called before start for '{name}'."
+            )
+
+        # Construct results
+        CPU_Times = collections.namedtuple("CPUtimes", list(cpu_times._fields))
+        cpu_times = CPU_Times(*np.subtract(cpu_times, self._start_cpu_times.pop(name)))
+        DiskIO = collections.namedtuple("DiskIO", list(disk_io._fields))
+        disk_io = DiskIO(*np.subtract(disk_io, self._start_disk_io.pop(name)))
+        memory = memory - self._start_memory.pop(name)
+
+        def bytes2human(num):
+            for unit in ["", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"]:
+                if abs(num) < 1024.0:
+                    return f"{num:3.1f}{unit}B"
+                num /= 1024.0
+            return f"{num:.1f}YiB"
+
+        memory = bytes2human(memory)
+
+        time_s = stop_time - self._start_time.pop(name)
+
+        if time_s < 0.1:
+            self._logger.info(
+                f"{name} ran for {time_s:.4f} < 0.1s, results might be inaccurate."
+            )
+
+        self._logger.info(
+            f"{name} ran for {time_s:.4f}s\n"
+            f"---------------------------------------------------------------------------------------------"
+            f"\n{cpu_times}\n"
+            f"Average CPU load: {cpu_percent}\n"
+            f"{disk_io}\n"
+            f"Change in (uss) memory: {memory}\n"
+            f"============================================================================================="
+        )
+
+    @property
+    def cpu_count(self):
+        """Number of cores available to this process."""
+        return len(self.cpu_affinity())
```

Comment:
From this docstring:

```
   cpu_percent :  float                                                                                                                                                                                  
             Process CPU utilization since `start` was called as percentage. Can be >100 if multiple threads run on                                                                                            
             different cores. See `PSUtil.cpu_count` for available cores.  
             ```
             
I think the idea was to allow for this to be used in more places.

Reply:
Sounds good.